### PR TITLE
enh(packaging): package centreon web using nfpm

### PR DESCRIPTION
**Pull request by kduret**
## Description

package centreon web using nfpm

**Fixes** MON-16861

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

### CODE REVIEW

**Review by sc979 — DISMISSED** [review]

**Review by jeremyjaouen — APPROVED** [review]

File: centreon/src/Core/Platform/Application/UseCase/UpdateVersions/UpdateVersionsException.php [hunk 1]
```diff
@@ -87,6 +87,14 @@ public static function errorWhenApplyingUpdate(
      */
     public static function errorWhenApplyingPostUpdate(\Throwable $ex): self
     {
-        return new self(_('An error occurred when applying post update actions'), 0, $ex);
+        return new self(
+            sprintf(
+                '%s: %s',
+                _('An error occurred when applying post update actions'),
+                $ex->getMessage()
```
Comment (callapa) [inline centreon/src/Core/Platform/Application/UseCase/UpdateVersions/UpdateVersionsException.php:94]: It's dangerous to expose error messages whose source is unknown.
That's what logs are for, among other things.

File: centreon/src/Core/Platform/Application/UseCase/UpdateVersions/UpdateVersionsException.php [hunk 2]
```diff
@@ -87,6 +87,14 @@ public static function errorWhenApplyingUpdate(
...
+            ),
+            0,
+            $ex
+        );
```
Comment (kduret) [inline centreon/src/Core/Platform/Application/UseCase/UpdateVersions/UpdateVersionsException.php:98]:
```suggestion
        return new self(_('An error occurred when applying post update actions'), 0, $ex);
```

**Review by jeremyjaouen — APPROVED** [review]

**Review by sc979 — DISMISSED** [review]

**Review by MedWassCh — APPROVED** [review]

File: centreon/www/include/configuration/configGenerate/xml/moveFiles.php [hunk 3]
```diff
@@ -305,8 +303,8 @@
                                 $host['name']
                             )
                         );
-                    } else {
-                        @chmod(rtrim($nagiosCfg["cfg_dir"], "/") . '/' . basename($filename), 0664);
+                    } elseif (posix_getuid() === fileowner($targetFilePath)) {
+                        @chmod($targetFilePath, 0664);
```
Comment (callapa) [inline centreon/www/include/configuration/configGenerate/xml/moveFiles.php:307]: What else ?

File: centreon/www/include/configuration/configGenerate/xml/moveFiles.php [hunk 4]
```diff
@@ -342,8 +341,8 @@
                                         $host['name']
                                     )
                                 );
-                            } else {
-                                @chmod(rtrim($centreonBrokerDirCfg, "/") . '/' . basename($fileCfg), 0664);
+                            } elseif (posix_getuid() === fileowner($targetFilePath)) {
```
Comment (callapa) [inline centreon/www/include/configuration/configGenerate/xml/moveFiles.php:344]: What else ?

**Review by sc979 — APPROVED** [review]

**Review by MedWassCh — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**